robottelo/common/ssh.py

```diff
@@ -20,19 +21,20 @@
 
 
 class SSHCommandResult(object):
-    """
-    Structure that returns in all ssh commands results.
-    """
+    """Structure that returns in all ssh commands results."""
 
-    def __init__(self, stdout=None, stderr=None,
-                 return_code=0, transform_csv=False):
+    def __init__(
+            self, stdout=None, stderr=None, return_code=0, output_format=None):
         self.stdout = stdout
         self.stderr = stderr
         self.return_code = return_code
-        self.transform_csv = transform_csv
-        #  Does not make sense to return suspicious CSV if ($? <> 0)
-        if transform_csv and self.return_code == 0:
-            self.stdout = csv_to_dictionary(stdout) if stdout else {}
+        self.output_format = output_format
+        #  Does not make sense to return suspicious output if ($? <> 0)
+        if output_format and self.return_code == 0:
+            if output_format == 'csv':
+                self.stdout = csv_to_dictionary(stdout) if stdout else {}
+            if output_format == 'json':
+                self.stdout = json.loads(stdout) if stdout else None
```

Comment:
Depending on the command a list or a single JSON object could be returned. I think is better to return None to indicate that it have not returned anything and not a false positive result.